content/08-mongodb/12-mongodb-transactions.mdx

```diff
@@ -252,3 +252,43 @@ Prisma's [MongoDB connector](https://www.prisma.io/docs/concepts/database-connec
 Join us in celebrating this milestone on April 25-29 with our [virtual MongoDB Launch week](https://www.prisma.io/mongodblaunch).  There will be exclusive workshops, opportunities for Atlas credits, great swag, and much more!
 
 </PrismaOutlinks>
+
+## FAQ
+
+<details><summary> Can you use MongoDB transactions in Node.js? </summary>
+
+Yes, MongoDB transactions can be used in Node.js. 
+
+MongoDB provides a helpful getting started guide for [“How to Use MongoDB Transaction in Node.js”](https://www.mongodb.com/blog/post/quick-start-nodejs--mongodb--how-to-implement-transactions).
+
+</details>
+
+<details><summary> What are MongoDB ACID transactions? </summary>
+
+ACID is an acronym for atomicity, consistency, isolation, and durability. All of these properties have to do with keeping the database in a valid state. 
+
+Because a transaction is a group of database operation being conducted together, ensuring database validity in the event of unexpected errors is critical. ACID compliant transactions secure against invalid database states.
+
+</details>
+
+<details><summary> How do you rollback transactions in MongoDB? </summary>
+
+[Read and write concern settings](https://www.mongodb.com/docs/manual/core/transactions/#transactions-and-write-concern) dictate whether you will be able to rollback a transaction in MongoDB. 
+
+For example, when you commit a transaction with a write concern of `w:1` , your transaction can be rolled back in the [case of a failover](https://www.mongodb.com/docs/manual/core/replica-set-rollbacks/).
```

Comment:
This was a bit confusing to me.  The situation described in [this section and its subsections](https://www.mongodb.com/docs/manual/core/replica-set-rollbacks/#avoid-replica-set-rollbacks) seems seems to point out that the data will be rolled back, but not intentionally.  My understanding of the situation is that basically, if you set `w:1`, you have the possibility of just straight up losing data committed in a transaction in the event of a failover.  In a sense that's "rolling back" the transaction, but since it's already been confirmed as written to the client, it's basically just losing that data and the subsections that follow seem to describe alternatives to avoid that scenario.

I guess the point I'm trying to make is that I think the intent of the question is "how to I intentionally roll back transactions", while the MongoDB docs on the subject seem to be written from the perspective of "how to I avoid having my data 'rolled back' by a failure".  I think MongoDB is kind of playing fast and loose with what "roll back" means in this case which is making it really difficult to try to answer the question using official sources.  AFAIK, MongoDB's transaction capabilities are more closely aligned with their idea of ["sessions"](https://www.mongodb.com/docs/manual/core/transactions/#transactions-and-sessions) (i.e. you can abort a session to roll back the operations you've performed within the session thus far).

Curious to hear your thoughts.

Reply:
I was very unsure how to answer this one as well. I like just going with the abort transaction route. It makes more sense to me as well.